### PR TITLE
Merge x-doc and occurrence setting, merge word based suggestion settings

### LINKED ISSUE #196353: Merge `editor.occurrencesHighlight` and `editor.multiDocumentOccurrencesHighlight` setting
Testing #196342

There is two settings which is confusing because what should happen when occurrencesHighlight is off but multiDocumentOccurrencesHighlight is on? A better solution would be to change `editor.occurrencesHighlight` to accept `true | false ` (as today) and a new value set of `"off" | "singleFile" | "acrossFiles"` (names to be defined)

### PR DESCRIPTION
**Pull request by Yoyokrazy**
Fixes: #196353

This mirrors the approach of word based suggestions and the associated settings there. No behavior changes at all with this, just much improved readability and the behavior more closely aligns with the setting descriptions/names.

Can also tweak and merge the settings into a single setting with multiple accepted value sets (ie boolean | enum), though to some effect I think that may reduce discoverability.

### CODE REVIEW

File: src/vs/editor/common/config/editorOptions.ts [hunk 1]
```diff
@@ -563,10 +563,10 @@ export interface IEditorOptions {
 	 */
 	occurrencesHighlight?: boolean;
```
Comment (rebornix) [inline src/vs/editor/common/config/editorOptions.ts:564]: I wonder if we want to extend the original setting to `true | false | 'singleFile' | 'multiFile'`, I personally would prefer a single setting if possible.

Comment (jrieken) [reply src/vs/editor/common/config/editorOptions.ts:564]: I agree. One setting is simpler than two. Word based suggestions should probably follow the lead

Comment (Yoyokrazy) [reply src/vs/editor/common/config/editorOptions.ts]: Merged both settings and went with a settings migration for both to keep things simple.

**Review by rebornix — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
@Yoyokrazy This PR breaks our extension's default settings -- what is the recommended way for us to handle this? There doesn't appear to be any migration or compatibility in place for settings values registered by extensions. I have PR at https://github.com/microsoft/vscode-cpptools/pull/11773/files that fixes it for 1.85.0, but that would regress old VS Code versions unless we increased our minimum supported VS Code version (causing users with 1.84.0 to not see our updates).

**Pull request merged**